### PR TITLE
build(lint): use strict eslint rules [EDS-310]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
-	{ languageOptions: { globals: globals.browser } },
-	pluginJs.configs.recommended,
-	...tseslint.configs.recommended,
+export default tseslint.config(
+	eslint.configs.recommended,
+	...tseslint.configs.strictTypeChecked,
+	{
+		languageOptions: {
+			globals: {
+				...globals.browser,
```
Comment (Nevoss) [inline eslint.config.mjs:11]: Why browser?

File: package.json [hunk 2]
```diff
@@ -16,6 +16,8 @@
 	"private": true,
 	"devDependencies": {
 		"@eslint/js": "^9.7.0",
+		"@types/eslint__js": "^8.42.3",
+		"@types/minimist": "^1.2.5",
```
Comment (Nevoss) [inline package.json:20]: minimist should be with zx, and what is `eslint__js`?

File: actions/setup-wp-env/create-wp-env-config.ts [hunk 3]
```diff
@@ -90,9 +90,18 @@ function getOptions(
...
+			const [from, to] = mapping.split(':').map((item) => item.trim());
+
+			if (from && to) {
+				acc.push([from, to]);
+			}
+
+			return acc;
+		},
+		[],
+	);
```
Comment (Nevoss) [inline actions/setup-wp-env/create-wp-env-config.ts:104]: Why? for the types?

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (StyleShit) [reply package.json:20]: for some reason zx doesn't ship minimist's types 🤷 
the eslint__js is types for the eslint configs
https://www.npmjs.com/package/@types/eslint__js

File: actions/setup-wp-env/create-wp-env-config.ts (hunk 3 above)
Comment (StyleShit) [reply actions/setup-wp-env/create-wp-env-config.ts:104]: yes

File: eslint.config.mjs (hunk 1 above)
Comment (StyleShit) [reply eslint.config.mjs]: removed

**Review by Nevoss — APPROVED** [review]

**Pull request merged**